### DISCUSSION TITLE
The problem of forced heat transfer exists in the heat exchanger

### DISCUSSION THREAD

**Discussion by sshhuuoo** [oemof/tespy#518] [Q&A]
Hello Sir, I found a problem when I made the system, the heat exchanger seems to have a problem of forced heat transfer, may I ask how to avoid it .I built a simple turbine expansion system where the temperature of carbon dioxide went from 25 to 125. But the water that gives him heat is only 80 degrees .This is obviously irrational .How can I fix this?
```python
from tespy.networks import Network
working_fluid = "CO2"

nw = Network(
    T_unit="C", p_unit="bar", h_unit="kJ / kg", m_unit="kg / s", s_unit="kJ / kgK"
)
from tespy.components import HeatExchanger
from tespy.components import SimpleHeatExchanger
from tespy.components import Sink
from tespy.components import Source
from tespy.components import Turbine

c_in = Source("refrigerant in")
amb_out = Sink("sink ambient")
c_in1 = Source("refrigerant in1")
tu =  Turbine("Turbine1")
cons = SimpleHeatExchanger("SimpleHeatExchanger")
cd = HeatExchanger("condenser")
amb_out1 = Sink("sink ambient1")

from tespy.connections import Connection

c0 = Connection(c_in, "out1",  cd, "in1", label="0")
c1 = Connection(cd, "out1", tu , "in1", label="1")
c2 = Connection(tu , "out1", cons, "in1", label="2")
c3 = Connection(cons, "out1", amb_out , "in1", label="3")

c10 = Connection(c_in1, "out1", cd, "in2", label="10")
c20 = Connection(cd, "out2",amb_out1, "in1", label="20")
nw.add_conns(c0, c1, c2, c3,  c10, c20)

cd.set_attr(pr1=1, pr2=1)
tu.set_attr(eta_s=0.85)
cons.set_attr(pr=1)

c0.set_attr(p=120,T=25,fluid={working_fluid: 1},m=1.543)

c2.set_attr(p=40,T=40)
c3.set_attr(T=30)

c10.set_attr(T=80,  fluid={"water": 1},m=1, p=1)

from tespy.connections import Bus

powergen = Bus("electrical power output")

powergen.add_comps(
    {"comp": tu, "char": 0.97, "base": "bus"},
)

nw.add_busses(powergen)

nw.solve(mode='design')
nw.print_results()
import matplotlib
matplotlib.rcParams['font.sans-serif'] = ['SimHei']     # 显示中文
# 为了坐标轴负号正常显示。matplotlib默认不支持中文，设置中文字体后，负号会显示异常。需要手动将坐标轴负号设为False才能正常显示负号。
matplotlib.rcParams['axes.unicode_minus'] = False

import matplotlib.pyplot as plt
x =[c0.s.val,c1.s.val,c2.s.val,c3.s.val]
y = [c0.T.val,c1.T.val,c2.T.val,c3.T.val]
plt.plot(x,y)
plt.title("温熵图")
plt.xlabel("s")
plt.ylabel("T")
for i,name in enumerate(["1","2","3","4"]):
    plt.annotate(name,(x[i],y[i]))
plt.show()
```

**Comment by sshhuuoo** [comment]
![image](https://github.com/oemof/tespy/assets/164991326/0b524d29-af2b-4946-a289-503f068c7f19)

**Reply by fwitte** [reply]
@sshhuuoo,

there are two issues, which cause the calculation not to converge and the results being infeasible:

- you specify both mass flows, that will lead to (due to the energy balance of the condenser) a forced outlet enthalpy on the water side, which seems to be out of the fluid property range of the water.
- you specify the turbine outlet temperature together with the pressure at inlet and outlet, which with the given isentropic efficiency leads to a turbine inlet temperature, which can never be met by the condenser (because it is actually higher than the water inlet temperature).

So, you might want to double check your temperature and pressure data because with your specifications, it is impossible to find something feasible here.

Best
